addon/components/mi-date.js

```diff
@@ -1,213 +1,242 @@
+                this.initFlatpickr();
+            }
+
+            run.scheduleOnce('afterRender', this, function () {
+                let {
+                    calendarButtonClass
+                } = configuration.getDateConfig();
+                set(this, 'calendarClass', calendarButtonClass);
+            });
+        }
     },
-    set(key, value) {
-      if (isNone(value)) {
-        return value;
-      }
-      let formattedDate = moment(value, 'YYYY-MM-DD', true);
-      if (!formattedDate.isValid()) {
-        set(this, 'value', null);
-      } else {
-        set(this, 'value', formattedDate.toDate());
-      }
-      set(this, 'mobileInputVisible', false);
-      this.get('onValueChanged')(this.get('value'));
-      return value;
-    }
-  }),
-
-  actions: {
-    actionCalendarButton() {
-      if ((this._getShowOn() === 'button') || (this._getShowOn() === 'both')) {
-        let calendar = get(this, 'pikadayCalendar');
-        if (isPresent(calendar)) {
-          calendar.show();
+
+    _getDateFormat() {
+        return getWithDefault(this, 'format', configuration.getDateConfig().format);
+    },
+
+    showCalendarButton: computed('showOn', function () {
+        let showOn = this._getShowOn();
+        if (showOn === 'button' || showOn === 'both') {
+            return true;
+        }
+        return false;
+    }),
+
+    // eslint-disable-next-line ember/no-on-calls-in-components
+    desktopTextColorObserver: on('init', observer('desktopValue', function () {
+        scheduleOnce('afterRender', this, function () {
+            if (isEmpty(get(this, 'desktopValue'))) {
+                $(this.element).find('.desktop-input').addClass('desktop-input-empty');
+            } else {
+                $(this.element).find('.desktop-input').removeClass('desktop-input-empty');
+            }
+        });
+    })),
+
+
+    desktopValue: computed('value', {
+        get() {
+            let value = get(this, 'value');
+            if (isNone(value)) {
+                return null;
+            }
+
+            //TODO newFormat
+            /*TODO [2:43 PM] Igor Kvasnička: tak potom by som urobil nejaku funkciu, ktorej na vstupe das flatpickr format a na vystupe bude format pre tu masku
+                [2:43 PM] Igor Kvasnička: staci ak budes podporovat len tie zakladne formaty: d, m , Y
+                [2:44 PM] Igor Kvasnička: cize na vstupe bude d.m.Y a na vystupe bude DD.MM.YYYY*/
+            let newFormat = (this._getDateFormat() === 'DD.MM.YYYY') ? 'd.m.Y' : 'Y-m-d';
+            return flatpickr.formatDate(value, newFormat);
+
+        },
+        set(key, value) {
+            //TODO newFormat
+            let newFormat = (this._getDateFormat() === 'DD.MM.YYYY') ? 'd.m.Y' : 'Y-m-d';
+
+            //TODO ech podmienka value.indexOf('_') === -1
+            // if (value.replace('_', "").length === flatpickr.formatDate(new Date(), newFormat).toString().length) {
+            if (value.indexOf('_') === -1) {
```

Comment:
This is a proper place to comment the code - what is the purpose of value.indexOf('_') === -1? When you look at this line in 2 months, you wouldn't remember the reason.